src/lib/update-lock.ts

```diff
@@ -101,5 +107,9 @@ export function lock(
 			.disposer(dispose);
 	};
 
-	return Bluebird.using(takeTheLock(), fn);
+	const disposer = takeTheLock();
+	if (disposer) {
```

Comment:
This changes it so that `fn` is never called in the case of disposer being falsey - is that intentional?